src/widget/wtrackmenu.h

```diff
@@ -96,23 +101,23 @@ class WTrackMenu : public QMenu {
     void slotAddToAutoDJReplace();
 
     // Cover
-    void slotCoverInfoSelected(const CoverInfoRelative &coverInfo);
+    void slotCoverInfoSelected(const CoverInfoRelative& coverInfo);
     void slotReloadCoverArt();
 
     // Library management
     void slotRemove();
     void slotHide();
     void slotUnhide();
     void slotPurge();
-    
-public:
-    signals:
-    void loadTrackToPlayer(TrackPointer pTrack, QString group,
-            bool play = false);
 
-private:
+  private:
+    TrackIdList getTrackIds() const;
+    TrackPointerList getTrackPointers() const;
+    TrackModel* getTrackModel() const;
+    QModelIndexList getTrackIndices() const;
```

Comment:
This has been done to avoid data redundancy. These methods take care of getting the list whether `trackModel` is available or not. If `trackModel` is available, indices are used to get pointers and ids, else pointers are used to get ids (getting indices does not make sense here).

Reply:
Okay, I understand what you mean. That only applies to `getTrackIds` and `getTrackPointers` though. `getTrackIndicies` and `getTrackModel` simply return the private members as-is without any additional logic.

Reply:
`getTrackIndices` is a trivial getter but exists to provide a function similar to `getTrackIds` and `getTrackPointers` thus maintaining consistency.
I will remove the `getTrackModel` function though and directly access `m_pTrackModel`.